cmd/ci-operator/main.go

```diff
@@ -53,9 +53,23 @@ func main() {
 	}
 }
 
+type stringSlice struct {
+	values []string
+}
+
+func (s *stringSlice) String() string {
+	return strings.Join(s.values, string(filepath.Separator))
+}
+
+func (s *stringSlice) Set(value string) error {
+	s.values = append(s.values, value)
+	return nil
+}
+
 type options struct {
-	rawBuildConfig string
-	templatePaths  []string
+	rawBuildConfig    string
+	templatePaths     []string
```

Comment:
Why should we pass paths on the filesystem for the config, template and secrets instead of just having the in-cluster client that the ci-operator has read the data from the apiserver itself?

Reply:
Because then you can't run it locally.

Reply:
Data is read from one place and written to one place.  We read from the filesystem and write to the server.  That's why pods have injection points - dynamic reading is unnecessary ceremony.